Cairo Desktop/CairoDesktop.Common/Localization/DisplayString.cs

```diff
@@ -1,720 +1,857 @@
+                    }
+                    else if (kvp.Value.StartsWith("it_"))
+                    {
+                        lang = Language.it_IT;
+                    }
+                    else if (kvp.Value.StartsWith("pt_"))
+                    {
+                        lang = Language.pt_BR;
+                    }
+                    else if (kvp.Value.StartsWith("sv_"))
+                    {
+                        lang = Language.sv_SE;
+                    }
+                    else if (kvp.Value.StartsWith("zh_"))
+                    {
+                        lang = Language.zh_CN;
+                    }
+                    else if (kvp.Value.StartsWith("cs_"))
+                    {
+                        lang = Language.cs_CZ;
+                    }
+                    else if (kvp.Value.StartsWith("de_"))
+                    {
+                        lang = Language.de_DE;
+                    }
+                    else if (kvp.Value.StartsWith("es_"))
+                    {
+                        lang = Language.es_ES;
+                    }
+                    else if (kvp.Value.StartsWith("nl_"))
+                    {
+                        lang = Language.nl_NL;
+                    }
+                    else if (kvp.Value.StartsWith("ru_"))
+                    {
+                        lang = Language.ru_RU;
+                    }
+                    else if (kvp.Value.StartsWith("pl_"))
+                    {
+                        lang = Language.pl_PL;
+                    }
+                    else if (kvp.Value.StartsWith("tr_"))
+                    {
+                        lang = Language.tr_TR;
+                    }
+                    else if (kvp.Value.StartsWith("ko_"))
+                    {
+                        lang = Language.ko_KR;
+                    }
+                    else if (kvp.Value.StartsWith("hu_"))
+                    {
+                        lang = Language.hu_HU;
+                    }
+                    if (lang != null)
+                    {
+                        for (int i = 0; i < lang.Count; i++)
+                        {
+                            sw.WriteLine(lang.ElementAt(i).Key + "=" + lang.ElementAt(i).Value);
+                        }
+                    }
+                }
+            }
+            //extract languages*/
+
+            if (System.IO.Directory.Exists(System.AppDomain.CurrentDomain.BaseDirectory+ "\\Localization"))
+            {
+                Languages.Clear();
+                string[] lngFiles = System.IO.Directory.GetFiles(System.AppDomain.CurrentDomain.BaseDirectory + "\\Localization", "*.lng");
+                for(int i=0; i< lngFiles.Length; i++) 
+                {
+                    Languages.Add(new KeyValuePair<string, string>(new System.IO.FileInfo(lngFiles[i]).Name.Replace(".lng", ""), new System.IO.FileInfo(lngFiles[i]).Name));
+                }
+            }

```

Comment:
I think enumerating the list of languages in the constructor here is going to be a bit too IO heavy for us to be doing on startup. I think we should instead only enumerate upon the first access of `Languages`.

Also, I think the filenames of each .lng file should be the locale ID (i.e. `en_US`) instead of the display name. We can get the display name from within the file when we populate `Languages`.

Reply:
I load list of languages here because I hijacked your system, your system also loads list here. I set file names as display string because then when we load list we don't need to open each file on the startup just to get name.